modules/islandora_entities_csv_import/includes/utilities.inc

```diff
@@ -68,10 +68,26 @@ function islandora_entities_read_csv($path) {
  */
 function islandora_entities_build_scholar_object($scholar, $collection_pid) {
 
+  module_load_include('inc', 'islandora', 'includes/utilities');
   $mads = islandora_entities_build_mads($scholar);
+  $namespace = 'islandora';
+  // Load the collection object and, upon success, get the appropriate
+  // namespace.
+  $collection_object = islandora_object_load($collection_pid);
```

Comment:
Ok, i feel we need to refactor something here. We are passing $collection_pid as a single value for the whole batch processor, not per object, right? That means that checking for each object over and over is an overkill in terms of processing (more if we can drop batches for non existing parent objects completely). I would move the whole "is the collection there and does it has a valid policy) out of islandora_entities_build_scholar_object and into whatever triggers the batch. Right now it seems to be only the form, but it could be also a drush script right? Maybe there is a need i did not understood here, but open to ideas and understanding this

Reply:
You are probably right. But this would mean to change the API of `islandora_entities_build_scholar_object()` to hand over the namespace as well, right? Is this wanted?